src/hooks/Tokens.ts

```diff
@@ -294,12 +294,22 @@ export const findCacheToken = (address: string) => {
   return cacheTokens[address] || cacheTokens[address.toLowerCase()]
 }
 
-export const fetchTokenByAddress = async (address: string, chainId: ChainId, signal?: AbortSignal) => {
+export const fetchTokenByAddress = async (address: string, chainId: ChainId, _signal?: AbortSignal) => {
   if (address === ZERO_ADDRESS) return NativeCurrencies[chainId]
   const findToken = findCacheToken(address)
   if (findToken && findToken.chainId === chainId) return findToken
-  const response = await axios.get(`${KS_SETTING_API}/v1/tokens?query=${address}&chainIds=${chainId}`, { signal })
-  const token = response?.data?.data?.tokens?.[0]
+  const tokenListRes = await store.dispatch(
+    ksSettingApi.endpoints.getTokenList.initiate({ chainId, addresses: address }),
+  )
+  let token = tokenListRes.data?.data.tokens[0]
+  if (!token) {
+    const importTokenRes = await store.dispatch(
+      ksSettingApi.endpoints.importToken.initiate([{ chainId: chainId.toString(), address }]),
+    )
+    if ('data' in importTokenRes) {
+      token = importTokenRes.data.data.tokens[0]?.data
```

Comment:
Might not need `?`.
```suggestion
      token = importTokenRes.data.data.tokens[0].data
```

Anyway I prefer `importTokenRes?.data?.data?.tokens?.[0]?.data` so no need `'data' in importTokenRes`.

Reply:
```suggestion
      token = importTokenRes?.data?.data?.tokens?.[0]?.data
```

Reply:
or add transformResponse [here](https://github.com/KyberNetwork/kyberswap-interface/blob/5f08d1099d2e0fe16eaf82a30851d5f2d675a6dd/src/services/ksSetting.ts#L121) 

and then
```token = importTokenRes?.tokens?.[0]?.data```

Reply:
> Might not need `?`.

It needed, `tokens[0]` could be undefined


> Anyway I prefer `importTokenRes?.data?.data?.tokens?.[0]?.data` so no need `'data' in importTokenRes`.

It's an issue with typescript not checking undefined, `importTokenRes` has type kind of unknown